pulsar-broker-auth-sasl/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderSasl.java

```diff
@@ -72,6 +75,10 @@ public class AuthenticationProviderSasl implements AuthenticationProvider {
 
     private JAASCredentialsContainer jaasCredentialsContainer;
     private String loginContextName;
+    private Cache<Long, AuthenticationState> authStates;
+
+    private static final String AUTHENTICATION_SASL_PREFIX = "authentication_sasl_prefix";
+    private static final long AUTHENTICATION_SASL_DEFAULT_EXPIRE_MS = 60_000;
```

Comment:
We have SASL configurations defined in [broker.conf](https://github.com/apache/pulsar/blob/master/conf/broker.conf#L885-L899), why not add a new one to the broker.conf?

Reply:
AuthenticationProviderSasl is an extension implementation, it is a bit inappropriate to put these in broker.conf.
SaslConstants is much more better ?

Reply:
Oh, good to know. My understanding is the SaslConstants is for SASL itself, but the newly introduced configuration is about how the broker expires the inflight SASL context to avoid OOM. 

@Technoboy- Do you know how to configure a similar behavior for Kafka?